### PR TITLE
Initial setup of sdk and created a method to fetch existing tranche configuration

### LINKED ISSUE #14: Fetch data from an existing tranche configuration
The vyper.js SDK offers methods to fetch a tranche configuration parameters given an account public key

### PR DESCRIPTION
**Pull request by iamakshat01**
- Closes #14

### CODE REVIEW

File: sdk/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export * from './Vyper';
+
+
+// for testing
```
Comment (thegreat-vanderlinde) [inline sdk/src/index.ts:4]: Can we use a more standard approach for testing? I'd like to introduce [Mocha](https://mochajs.org/), we're using it in all our project. At [this link](https://mochajs.org/#getting-started) you have a getting started guide

File: sdk/src/types/Tranche.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as anchor from "@project-serum/anchor";
+
+
+export interface LastUpdateType {
```
Comment (thegreat-vanderlinde) [inline sdk/src/types/Tranche.ts:4]: I don't get why we need both an interface and a class for the same type, can't we use just the class object?

File: sdk/src/TrancheConfig.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import { PublicKey } from "@solana/web3.js";
+import { TrancheDataType, ReserveFairValueType, SlotTrackingType, LastUpdateType, TrancheFairValueType } from './types/Tranche'
+
+export class TrancheConfig {
```
Comment (thegreat-vanderlinde) [inline sdk/src/TrancheConfig.ts:4]: can we have a single class per file? for example:
TrancheConfig.ts -> `export class TrancheConfig`
TrancheData.ts -> `export class TrancheData`

**Review by thegreat-vanderlinde — CHANGES_REQUESTED** [review]
the overall structure seems super clean, can we discuss the outlined points?

File: sdk/src/index.ts (hunk 1 above)
Comment (iamakshat01) [reply sdk/src/index.ts:4]: Seems good, will setup mocha

File: sdk/src/types/Tranche.ts (hunk 2 above)
Comment (iamakshat01) [reply sdk/src/types/Tranche.ts:4]: I tried that but there are some nested fields in TrancheConfig which are of type anchor.BN such as LastUpdate. For those I took them as input in create function and converted them to number there. For that input argument type  I created a separate interface with anchor.BN.

File: sdk/src/TrancheConfig.ts (hunk 3 above)
Comment (iamakshat01) [reply sdk/src/TrancheConfig.ts:4]: Sounds good, doing it.

File: sdk/src/types/Tranche.ts (hunk 2 above)
Comment (thegreat-vanderlinde) [reply sdk/src/types/Tranche.ts:4]: mmh ok, I see. maybe we can structure our code with no anchor related stuff (i.e. `anchor.BN`) and when is time to convert fetched class from the chain to our classes we convert any non standard data type to a standard one. You can use this pattern as a reference (should have access to pvt repo):
https://github.com/vyper-protocol/vyper-vaults/blob/main/app/sdk/src/VaultData.ts

File: sdk/src/TrancheData.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { TrancheFairValue } from "./TrancheFairValue";
+
+export class TrancheData {
+    depositedQuantity: number[];
+    feeToCollectQuantity: number;
+    reserveFairValue: ReserveFairValue;
+    trancheFairValue: TrancheFairValue;
+    haltFlags: number;
+    ownerRestrictedIx: number;
+    padding: number[];
```
Comment (thegreat-vanderlinde) [inline sdk/src/TrancheData.ts:11]: padding can be omitted

File: sdk/src/TrancheData.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { ReserveFairValue } from "./ReserveFairValue";
+import { TrancheFairValue } from "./TrancheFairValue";
+
+export class TrancheData {
+    depositedQuantity: number[];
+    feeToCollectQuantity: number;
+    reserveFairValue: ReserveFairValue;
+    trancheFairValue: TrancheFairValue;
+    haltFlags: number;
```
Comment (thegreat-vanderlinde) [inline sdk/src/TrancheData.ts:9]: can we have like an enum for `haltFlags` and `ownerRestrictedIx`? here you can find some refs: https://patrickdesjardins.com/blog/how-to-set-and-read-bitwise-enum-values-in-typescript

**Review by thegreat-vanderlinde — CHANGES_REQUESTED** [review]
we just need to fix the enum commented and we're good to merge the pr

File: sdk/src/TrancheData.ts (hunk 4 above)
Comment (iamakshat01) [reply sdk/src/TrancheData.ts]: Done

File: sdk/src/TrancheData.ts (hunk 5 above)
Comment (iamakshat01) [reply sdk/src/TrancheData.ts]: Created enums for them

**Review by thegreat-vanderlinde — APPROVED** [review]

**Pull request merged**